optimum/intel/openvino/modeling_decoder.py

```diff
@@ -271,10 +330,22 @@ def forward(
 
         inputs = {}
         if past_key_values is not None:
-            # Flatten the past_key_values
-            past_key_values = tuple(
-                past_key_value for pkv_per_layer in past_key_values for past_key_value in pkv_per_layer
-            )
+            if len(self.key_value_input_names):
```

Comment:
if I right remember if use_cache = True during model loading and key_value_input_names are empty, model creating will fail with error that model does not support this model, so possibly this check is not necessary here

Reply:
Yes, self.use_cache is set based on the presence of pkv in model inputs here https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/openvino/modeling_decoder.py#L122 and there is a check a few lines later that throws a ValueError if does not match with the use_cache value a user provides.

Reply:
Updated.